Feature/view sensor statues

SMiller-0 opened the pull request:
The branch includes:
- Basic Sensor status page displaying the active status of each sensor.
   - This page is not final as I am undecided on what other details to display on this page.
- Timer service used to create timers.
- Refresh service used to automatically refresh the page (frequency of refresh set by timer service).
- All Sensor Page changed to refresh on navigation to ensure changes are reflected.

Note: If you want to test the auto-refresh, you can run the app with the sensor status page open, then use Azure Data Studio to add a new sensor or change the is_active setting of a sensor and see it automatically update in the app.

Next Steps:
- Add more details to Sensor Status Page
- Add comments and documentation

Alanna-Mc reviewed (commented):
The sensor status feature works as expected. I tested the auto-refresh functionality by updating the is_active value directly in the database, and the UI updated correctly. The code is well-structured and fits with the existing MVVM approach. There is also clear internal documentation, which made the classes and methods easier to follow and understand.

While testing, I did run into a couple of issues. However, these may be environment-related on my side, as I also checked out the develop branch and saw the same problems (even though they were working previously):

- The "View All Sensors" filter doesn’t seem to be applying properly when I select a filter.

- Editing a sensor is currently throwing an exception, and it looks like the selected sensor details are no longer being passed correctly to the sensor page.  

Let me know if you experience these same issues.

<img width="419" alt="Filter" src="https://github.com/user-attachments/assets/5c4b65f7-0d1a-4422-8cfb-07f5824e0207" />

SMiller-0 commented:
I was able to recreate these issues on my end. It seems the edit issue stems from me accidently editing one of the routes in NavigationService.cs. I have reverted this back to the original and it has resolved the issue on my end. I have also fixed the bug with the filter.

Alanna-Mc reviewed (approved):
I have run the application again and everything is running as expected, all errors resolved.
